frontend/cypress/e2e/filtro.spec.js

```diff
@@ -0,0 +1,24 @@
+describe('Filtro de Clientes - Mantém Filtros ao Voltar dos Detalhes', () => {
+    beforeEach(() => {
+      cy.visit('http://localhost:3000');
+  
+      cy.get('[data-testid="size-filter"]').select('Medium');
+      cy.get('[data-testid="industry-filter"]').select('Technology');
+  
+      cy.get('[data-testid="table"]').should('exist');
+      cy.get('tbody tr').should('have.length.greaterThan', 0);
+    });
+  
+    it('deve manter os filtros aplicados após voltar da página de detalhes do cliente', () => {
+      cy.get('tbody tr').first().find('button[aria-label^="View company"]').click();
+  
+      cy.get('h2').contains('Customer Details');
+      
+      cy.contains('button', 'Back').click({ force: true });
```

Comment:
Evite forçar um clique.

Isso pode esconder problemas reais na aplicação

Testes _end-to-end_ servem para simular o uso da aplicação como um usuário real.

Se um clique está tendo que ser forçado, isso significa que o elemento não está em um estado clicável (quem sabe haja um elemento na frente dele).

Se um elemento não está num estado clicável, um usuário real não conseguiria clicar nele, portanto, ao forçar o clique, o teste não estaria simulando o correto uso da aplicação.

A propósito, por qual motivo você forçou o clique?

Reply:
Estava ocorrendo um erro dizendo que havia um elemento na frente, quando na verdade não havia. Se bem me lembro esse elemento em questão era a modal dos cookies e, mesmo eu tentando escondê-lo ou fechá-lo no beforeEach, ainda ocorria o erro. Como eu queria apenas seguir o fluxo do teste, encontrei esse tipo de contorno (não muito convencional, pelo visto)